dojo/tools/veracode/json_parser.py

```diff
@@ -0,0 +1,374 @@
+import json
+import re
+from cvss import CVSS3
+
+from dojo.models import Finding
+from dojo.models import Endpoint
+
+
+class VeracodeJSONParser(object):
+    """This parser is written for Veracode REST Findings.
+
+    API endpoints to use: https://docs.veracode.com/r/c_findings_v2_examples
+
+    Example: curl <endpoint> | jq "{findings}"
+
+    This should convert the format into something like this:
+    {
+        "findings": [
+            {
+                ...
+            },
+            ...
+        ]
+    }
+    """
+
+    severity_mapping = {
+        1: "Info",
+        2: "Low",
+        3: "Medium",
+        4: "High",
+        5: "Critical",
+    }
+
+    exploitability_mapping = {
+        -2: "Very Unlikely",
+        -1: "Unlikely",
+        0: "Neutral",
+        1: "Likely",
+        2: "Very Likely",
+    }
+
+    # This mapping was found here: https://docs.veracode.com/r/c_integrated_license_agent
+    license_mapping = {
+        1: ("Unrecognized", "Unrecognized indicates that no license was found for the component. However, this does not indicate that there is no risk associated with the license."),
+        2: ("Low", "Low-risk licenses are typically permissive licenses that require you to preserve the copyright and license notices, but allow distribution under different terms without disclosing source code."),
+        3: ("Medium", "Medium-risk licenses are typically weak copyleft licenses that require you to preserve the copyright and license notices, and require distributors to make the source code of the component and any modifications under the same terms."),
+        4: ("High", "High-risk licenses are typically strong copyleft licenses that require you to preserve the copyright and license notices, and require distributors to make the source code of the component and any modifications under the same terms."),
+        5: ("Non OSS", "Non-OSS indicates that this file could be subject to commercial license terms. If so, you should refer to your applicable license agreement with such vendor for additional information."),
+    }
+
+    def get_findings(self, json_output, test):
+        findings = []
+        if json_output:
+            json_data = self.parse_json(json_output)
+            findings += self.get_items(json_data, test)
+        return findings
+
+    def parse_json(self, json_output):
+        data = json_output.read()
+        try:
+            json_findings = json.loads(str(data, 'utf-8'))
+        except Exception:
+            json_findings = json.loads(data)
+
+        return json_findings
```

Comment:
```suggestion
    def get_findings(self, json_output, test):
        findings = []
        if json_output:
            json_data = json.load(json_output)
            findings += self.get_items(json_data, test)
        return findings
```
`json` module do a better job than us to support UTF8/UTF16 shenanigans.